Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var center = [this._object.x + this._radius, this._object.y + this._radius];
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.hitBoxes;
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.flat(1);
+  var vertices = polygons.map(function (poly) {
+    return poly.vertices;
+  }).flat(1);
+
+  function _calculatePOI(angle) {
+    var minDist = halfOfDiag;
+    var minPOI = [];
```

Comment:
This is a point, so prefer to declare it as such for readibility and to help the JS engine by never modifying its length:

```
var minPOI = [0, 0];
```
then later:
```
minPOI[0] = poi.closeX
minPOI[1] = poi.closeY
```

Reply:
Actually, the default point should be `var minPOI = [center[0] + halfOfDiag * Math.cos(angle), center[1] + halfOfDiag * Math.sin(angle)];` (corresponding to a squared distance of `minSqDist = halfOfDialog*halfOfDialog`)

Because it's reused a lot, store it once:
```
const targetX = center[0] + halfOfDiag * Math.cos(angle);
const targetY = center[1] + halfOfDiag * Math.cos(angle);
```

and reuse it in `raycastTest`.